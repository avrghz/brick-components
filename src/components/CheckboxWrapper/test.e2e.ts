import { newE2EPage, E2EPage, E2EElement } from '@stencil/core/testing'

describe('<bk-checkbox-wrapper/>', () => {
    let page: E2EPage
    let component: E2EElement

    const getLabel = async () => await page.find('bk-checkbox-wrapper .bk-checkbox')

    beforeEach(async () => {
        page = await newE2EPage()
        await page.setContent(`<bk-checkbox-wrapper label="Option">
            <input type="checkbox" aria-hidden="false" value="option" name="options" />
        </bk-checkbox-wrapper>`)
        component = await page.find('bk-checkbox-wrapper')
    })

    it('should render', async () => {
        expect(component).toMatchSnapshot('snapshot__basic')
    })

    describe('@Prop indeterminate', () => {
        beforeEach(async () => {
            component.setProperty('indeterminate', true)
            await page.waitForChanges()
        })

        it('should render in indeterminate state', async () => {
            expect(await getLabel()).toHaveClass('is-indeterminate')
        })

        it('should disable indeterminate state on click', async () => {
            await component.click()
            await page.waitForChanges()
            expect(await getLabel()).not.toHaveClass('is-indeterminate')
        })
    })
})
