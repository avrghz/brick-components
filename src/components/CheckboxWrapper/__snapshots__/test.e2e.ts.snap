// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bk-checkbox-wrapper/> should render: snapshot__basic 1`] = `
<bk-checkbox-wrapper class="hydrated sc-bk-checkbox-wrapper-h sc-bk-checkbox-wrapper-s" label="Option">
  <!---->
  <label class="bk-checkbox sc-bk-checkbox-wrapper sc-bk-checkbox-wrapper-s">
    <input aria-hidden="false" name="options" type="checkbox" value="option">
    <span class="bk-checkbox__input sc-bk-checkbox-wrapper">
      <span class="bk-checkbox__inner sc-bk-checkbox-wrapper"></span>
    </span>
    <span class="bk-checkbox__label sc-bk-checkbox-wrapper">
      Option
    </span>
  </label>
</bk-checkbox-wrapper>
`;
